src/fswAlgorithms/attDetermination/sunlineSRuKF/sunlineSRuKF.cpp

```diff
@@ -0,0 +1,280 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, Laboratory for Atmospheric and Space Physics, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include "sunlineSRuKF.h"
+
+SunlineSRuKF::SunlineSRuKF() = default;
+
+SunlineSRuKF::~SunlineSRuKF() = default;
+
+/*! Initialize C-wrapped output messages */
+void SunlineSRuKF::SelfInit(){
+    NavAttMsg_C_init(&this->navAttOutMsgC);
+}
+
+/*! Reset the flyby OD filter to an initial state and
```

Comment:
:small_red_triangle_down: Check all comments and function headers are accurate and descriptive.

Reply:
Done